bindings/python/shellcode.py

```diff
@@ -65,15 +95,36 @@ def hook_intr(uc, intno, user_data):
         print(">>> 0x%x: interrupt 0x%x, EAX = 0x%x" %(eip, intno, eax))
 
 
-def hook_syscall(mu, user_data):
-    rax = mu.reg_read(UC_X86_REG_RAX)
-    print(">>> got SYSCALL with RAX = 0x%x" %(rax))
+def hook_syscall32(mu, user_data):
+    eax = mu.reg_read(UC_X86_REG_EAX)
+    ebx = mu.reg_read(UC_X86_REG_EBX)
+    print(">>> got SYSCALL with EAX = 0x%x" %(eax))
     mu.emu_stop()
 
+def hook_syscall64(mu, user_data):
+    rax = mu.reg_read(UC_X86_REG_RAX)
+    rdi = mu.reg_read(UC_X86_REG_RDI)
+
+    print(">>> got SYSCALL with RAX = %d" %(rax))
+    
+    if rax == 59:    #sys_execve
+        filename = read_string(mu, rdi)
+        print(">>> SYS_EXECV filename=%s" % filename)
+
+    else:
+        print("=" * 20)
+        print(">>> Syscall Found 0x%x: interrupt 0x%x, RAX = 0x%x" %(eip, intno, eax))
```

Comment:
this line should be fixed: after copy/paste, we no longer have eip, intno & eax here?

Reply:
updated